### PR TITLE
Make use of github.com/mattermost/mattermost/server/public

### PR DESCRIPTION
**Pull request by hanzei**
This PR also updated all go dependencies.

Fixes https://github.com/matterpoll/matterpoll/issues/500

## Summary by CodeRabbit

Based on the comprehensive summary of changes, here are the updated release notes:

- **New Features**
  - Added log retrieval and monitoring capabilities for plugins via CLI commands
  - Enhanced plugin management with new log-related functionality

- **Dependency Updates**
  - Updated Go version to 1.22
  - Migrated import paths from `mattermost-server/v6` to `mattermost/server/public`
  - Updated numerous package dependencies to their latest versions

- **Improvements**
  - Introduced context handling for API calls to improve timeout and cancellation management
  - Streamlined import paths for better package organization

- **Chores**
  - Refactored import statements across multiple files
  - Updated test utilities and mock implementations

- **Minor Changes**
  - Removed an edge case test for empty message localization
  - Minor adjustments to mock argument generation in test utilities

### CODE REVIEW

File: build/pluginctl/logs.go
Comment (hanzei) [inline build/pluginctl/logs.go:1]: See https://github.com/mattermost/mattermost-plugin-starter-template/pull/193

File: go.mod [hunk 1]
```diff
@@ -1,74 +1,61 @@
...
+toolchain go1.22.8
 
 require (
 	github.com/blang/semver/v4 v4.0.0
-	github.com/gorilla/mux v1.8.0
-	github.com/mattermost/mattermost-plugin-api v0.0.27
-	github.com/mattermost/mattermost-server/v6 v6.5.2
-	github.com/nicksnyder/go-i18n/v2 v2.2.0
+	github.com/gorilla/mux v1.8.1
+	github.com/mattermost/mattermost/server/public v0.1.9
```
Comment (hanzei) [inline go.mod:10]: Welcome to the future :sunglasses:

**Comment by kaakaa** [comment]
Thanks @hanzei! This PR is important to keep up with the latest MM.

[CI error](https://github.com/matterpoll/matterpoll/actions/runs/12633961892/job/35200698146?pr=571#step:4:722) would be caused by the fact that go-i18n has been updated so that [`l.LocalizeMessage`](https://github.com/matterpoll/matterpoll/blob/master/server/utils/i18n.go#L85) returns an error if the argument `m *i18n.Message` is empty. But I think that `TestLocalizeDefaultMessage/empty message` test is meaningless, so we can remove the failing test.

File: go.mod [hunk 2]
```diff
@@ -1,74 +1,61 @@
...
+	github.com/nicksnyder/go-i18n/v2 v2.4.1
```
[the remark this answers is not in the document]
Comment (hanzei) [reply go.mod:11]: As mentioned in https://github.com/matterpoll/matterpoll/pull/571#issuecomment-2599768452, I think it's fine to just remove the test case.

**Comment by hanzei** [comment]
@kaakaa PTAL

**Review by kaakaa — APPROVED** [review]
👍
